src/main/resources/db/changelog/db.changelog-master.yaml

```diff
@@ -419,3 +419,32 @@ databaseChangeLog:
       changes:
         - tagDatabase:
             tag: version_0.1.0
+  - changeSet:
+      id: sneaky-view
+      author: ben.warfield@usds.dhs.gov
+      comment: This had better not make it directly into prod.
+      runOnChange: true
+      changes:
+        - sql:
+            sql: drop view if exists ${database.defaultSchemaName}.sneaky_view
+        - createView:
+            viewName: sneaky_view
+            selectQuery: |
```

Comment:
This feels wrong. To me it seems that a select query like this would exist somewhere within the java logic and not in a migrations file.

What happens if we update one of the columns referenced in this query? Would we have to update this in a new changeset?

Reply:
I have mixed feelings about it myself. To cut to the TL;DR, though: for the moment, and unless somebody makes a _really_ compelling argument to the contrary, we're using Hibernate as our JPA implementation, and while Hibernate supports having a view definition in the Java source using the [Subselect](https://docs.jboss.org/hibernate/orm/5.2/javadocs/org/hibernate/annotations/Subselect.html) annotation, that annotation does _not_ [support the {h-schema} interpolation](https://hibernate.atlassian.net/browse/HHH-7913) that our existing named native queries do, so effectively we can't use it.

So, as a practical matter, I think we're stuck with this approach until Hibernate 6 is out _and_ we care enough to come back and change this, which is ... I'm not sure that's an actual milestone?

To the practical point: yes, if we update a table used in this view, we would have to update the view—though if we update the underlying table we're already running migrations and paying that cost, so I'm not sure how much of an issue that is? On the other hand, as a general note of caution, it's probably worth remembering that creating views creates one more coupling point that you have to maintain, so to the extent that we can bias the next round of development toward Java-side changes rather than expansions of views, that's probably worth doing.